src/core/connection.h

```diff
@@ -1088,7 +1093,15 @@ QuicConnAddRef(
     UNREFERENCED_PARAMETER(Ref);
 #endif
 
-    CxPlatRefIncrement(&Connection->RefCount);
+    if (!Connection->State.CleanupStarted) {
```

Comment:
I think we should have this check in the caller instead.
There is only a single place in the code where we should be trying to add a ref to a connection that reached a refcount of zero.
Any other place would be a pretty bad bug I think we should catch even in release builds.